plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3select/TestS3SelectPushdown.java

```diff
@@ -43,4 +102,112 @@ public void testIsCompressionCodecSupported()
+                Optional.empty(), Optional.empty(), OptionalLong.empty());
+        assertFalse(S3SelectPushdown.shouldEnablePushdownForTable(session, newTable, "s3://fakeBucket/fakeObject", Optional.empty()));
+
+        Partition newPartition = new Partition("db", "table", emptyList(), newStorage, singletonList(column), emptyMap());
+        assertFalse(S3SelectPushdown.shouldEnablePushdownForTable(session, newTable, "s3://fakeBucket/fakeObject", Optional.of(newPartition)));
+
+        newStorage = Storage.builder()
+                .setStorageFormat(StorageFormat.create(LazySimpleSerDe.class.getName(), TextInputFormat.class.getName(), "outputFormat"))
+                .setLocation("location")
+                .build();
+        newTable = new Table("db", "table", Optional.of("owner"),
+                "type", newStorage, singletonList(column), emptyList(), Map.of(SKIP_HEADER_COUNT_KEY, "1"),
+                Optional.empty(), Optional.empty(), OptionalLong.empty());
+        assertFalse(S3SelectPushdown.shouldEnablePushdownForTable(session, newTable, "s3://fakeBucket/fakeObject", Optional.empty()));
+
+        newTable = new Table("db", "table", Optional.of("owner"),
+                "type", newStorage, singletonList(column), emptyList(), Map.of(SKIP_FOOTER_COUNT_KEY, "1"),
+                Optional.empty(), Optional.empty(), OptionalLong.empty());
+        assertFalse(S3SelectPushdown.shouldEnablePushdownForTable(session, newTable, "s3://fakeBucket/fakeObject", Optional.empty()));
+    }
+
+    @Test
+    public void testShouldNotEnableSelectPushdownWhenColumnTypesAreNotSupported()
+    {
+        Column newColumn = new Column("column", HIVE_BINARY, Optional.empty());
+        Table newTable = new Table("db", "table", Optional.of("owner"),
+                "type", storage, singletonList(newColumn), emptyList(), emptyMap(),
+                Optional.empty(), Optional.empty(), OptionalLong.empty());
+        assertFalse(S3SelectPushdown.shouldEnablePushdownForTable(session, newTable, "s3://fakeBucket/fakeObject", Optional.empty()));
+
+        Partition newPartition = new Partition("db", "table", emptyList(), storage, singletonList(newColumn), emptyMap());
+        assertFalse(S3SelectPushdown.shouldEnablePushdownForTable(session, newTable, "s3://fakeBucket/fakeObject", Optional.of(newPartition)));
+    }
+
+    @Test
+    public void testShouldEnableSplits()
+    {
+        // Uncompressed CSV
+        assertTrue(shouldEnableSplits(true, schema, inputFormat, new Path("s3://fakeBucket/fakeObject.csv")));
+        // Pushdown disabled
+        assertTrue(shouldEnableSplits(false, schema, inputFormat, new Path("s3://fakeBucket/fakeObject.csv")));
+    }
+
+    @Test
+    public void testShouldNotEnableSplits()
+    {
+        // Compressed file
+        assertFalse(shouldEnableSplits(true, schema, inputFormat, new Path("s3://fakeBucket/fakeObject.gz")));
+        // JSON
+        schema.setProperty(SERIALIZATION_LIB, JsonSerDe.class.getName());
```

Comment:
The tests can be running in parallel. Tests shouldn't modify shared data structures. You can create a new one every time you need it. Those should be cheap to create.